etcdserver/api/etcdhttp/peer.go

```diff
@@ -34,23 +34,32 @@ import (
 const (
 	peerMembersPath         = "/members"
 	peerMemberPromotePrefix = "/members/promote/"
+	downgradeStatusPath     = "/downgrade"
+	downgradeEnabledPath    = "/downgrade/enabled"
 )
 
 // NewPeerHandler generates an http.Handler to handle etcd peer requests.
 func NewPeerHandler(lg *zap.Logger, s etcdserver.ServerPeer) http.Handler {
 	return newPeerHandler(lg, s, s.RaftHandler(), s.LeaseHandler())
 }
 
-func newPeerHandler(lg *zap.Logger, s etcdserver.Server, raftHandler http.Handler, leaseHandler http.Handler) http.Handler {
+func newPeerHandler(lg *zap.Logger, s etcdserver.ServerPeer, raftHandler http.Handler, leaseHandler http.Handler) http.Handler {
```

Comment:
Maybe create a new interface `ServerDowngradable` instead of modifying the existing Cluster interface?